test/src/sdk-harness/test_projects/storage_map/mod.rs

```diff
@@ -0,0 +1,477 @@
+use fuels::prelude::*;
+use fuels_abigen_macro::abigen;
+
+abigen!(
+    TestStorageMapContract,
+    "test_projects/storage_map/out/debug/storage_map-abi.json",
+);
+
+async fn test_storage_map_instance() -> TestStorageMapContract {
+    let wallet = launch_provider_and_get_single_wallet().await;
+    let id = Contract::deploy(
+        "test_projects/storage_map/out/debug/storage_map.bin",
+        &wallet,
+        TxParameters::default(),
+    )
+    .await
+    .unwrap();
+
+    TestStorageMapContract::new(id.to_string(), wallet)
+}
+
+#[tokio::test]
+async fn can_insert_and_get() {
```

Comment:
This test is way too large and difficult to follow. Split it into multiple functions. I would also consider assigning variables and passing them in into the "set" and "get" so that you can change something in 1 place and have the test continue to work rather than remembering to change it in both calls. If the functions are small enough then this may not be needed

Reply:
Fully agree. The reason I decided to go with one big function is to show that you can defined many storage maps, insert into all of them separately, and then be able to read from all of them with no collision. That being said, I can probably showcase that with a much simpler large (but much smaller than this) test and let some other smaller tests check for individual case.